sdk/core/core-amqp/src/errors.ts

```diff
@@ -548,46 +616,42 @@ function isBrowserWebsocketError(err: any): boolean {
 }
 
 /**
- * Translates the AQMP error received at the protocol layer or a generic Error into a MessagingError.
+ * Translates the AQMP error received at the protocol layer or a SystemError into a MessagingError.
+ * All other errors are returned unaltered.
  *
  * @param {AmqpError} err The amqp error that was received.
  * @returns {MessagingError} MessagingError object.
  */
-export function translate(err: AmqpError | Error): MessagingError {
+export function translate(err: AmqpError | Error): MessagingError | Error {
   if ((err as MessagingError).translated) {
```

Comment:
Do we need `translated` at all?
Can `err.name === "MessagingError"` or `err instanceOf MessagingError` replace `(err as MessagingError).translated`?

Reply:
I like the idea of using err.name now that we've standardized. I'll update!